### PR TITLE
add registration for locations

### PR DESCRIPTION
**Pull request by AndrewCLu**

### CODE REVIEW

File: src/pages/tap.tsx [hunk 1]
```diff
@@ -23,6 +26,15 @@ export default function Tap() {
     [router]
   );
 
+  // Save the newly tapped location to local storage and redirect to their profile
+  const processLocationTap = useCallback(
```
Comment (RiverRuby) [inline src/pages/tap.tsx:30]: Why does this need to be a callback? Not blocking merge, just curious

File: src/pages/locations/[id].tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  const { id } = router.query;
+  const [location, setLocation] = useState<Location>();
+  const [locationSignature, setLocationSignature] =
+    useState<LocationSignature>();
+
+  useEffect(() => {
+    const fetchLocation = async () => {
+      if (typeof id === "string") {
+        try {
+          const response = await fetch(`/api/location/${id}`);
```
Comment (RiverRuby) [inline src/pages/locations/[id].tsx:21]: Any reason to protect this endpoint with a ZK PoK of signature? like are any locations gonna be private on purpose for scavenger hunts? Def doesn't need to happen in this PR but if you think it's important we should add an issue

File: src/pages/tap.tsx (hunk 1 above)
Comment (RiverRuby) [reply src/pages/tap.tsx:30]: Ah I see, so this works when you redirect to LoginForm?

**Review by RiverRuby — APPROVED** [review]
LGTM! I thought adding location tap would be a much smaller change but all of this code seems necessary & well-written. great work!!

File: src/pages/tap.tsx (hunk 1 above)
Comment (AndrewCLu) [reply src/pages/tap.tsx:30]: yup!

File: src/pages/locations/[id].tsx (hunk 2 above)
Comment (AndrewCLu) [reply src/pages/locations/[id].tsx:21]: There could be cases where this is useful, such as a scavenger hunt - i think in that case we can always protect the individual location with a PoK. If we plan on having such an activation its worth keeping in mind but for now doesn't seem to warrant the issue imo

**Pull request merged**